src/mos_editor.c

```diff
@@ -547,49 +547,24 @@ UINT24 mos_EDITLINE(char * buffer, int bufferLength, UINT8 flags) {
 		}
 
 		if (enableHistory) {
+			BOOL lineChanged = FALSE;
 			switch (historyAction) {
 				case 1: { // Push new item to stack
-					if (len > 0) {		// If there is data in the buffer
-						// If we're at the end of the history, then we need to shift all our entries up by one
-						if (history_size == (cmd_historyDepth - 1)) {
-							int i;
-							for(i = 0; i < history_size; i++) {
-								strncpy(cmd_history[i], cmd_history[i+1], cmd_historyWidth);
-							}
-							history_size--;
-						}
-						strncpy(cmd_history[history_size++], buffer, cmd_historyWidth);	// Save in the history
-						history_no = history_size;
-					}
+					editHistoryPush(buffer);

```

Comment:
this switch got refactored to use functions for each history action so as to allow those functions to more easily be refactored to use a different way to store history items